### PR TITLE
COTECH-612 Add Vimeo takeover option to the video wrappers

### PR DESCRIPTION
**Pull request by nandy-andy**

### CODE REVIEW

File: src/vimeo/players/shared/CloseButton.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	pointer-events: initial;
+	height: ${(props) => (props.deviceType === 'desktop' ? '24px' : '36px')};
+	width: ${(props) => (props.deviceType === 'desktop' ? '24px' : '36px')};
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	opacity: 0.98;
+	position: absolute;
+	right: -3px;
+	right: ${(props) => (props.deviceType === 'desktop' ? '-3px' : '-6px')};
```
Comment (pbartkowiak-fandom) [inline src/vimeo/players/shared/CloseButton.tsx:18]: Duplicated property.

File: src/vimeo/players/shared/CloseButton.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import styled from 'styled-components';
+import WDSVariables from '@fandom-frontend/design-system/dist/variables.json';
+import IconCrossTiny from '@fandom-frontend/react-common/dist/icons/IconCrossTiny';
+import { VimeoArticleVideoPlayerTrackingProps } from 'vimeo/types';
+
+const CloseWrapper = styled.div<VimeoArticleVideoPlayerTrackingProps>`
+	cursor: pointer;
+	pointer-events: initial;
+	height: ${(props) => (props.deviceType === 'desktop' ? '24px' : '36px')};
+	width: ${(props) => (props.deviceType === 'desktop' ? '24px' : '36px')};
```
Comment (pbartkowiak-fandom) [inline src/vimeo/players/shared/CloseButton.tsx:11]:
```suggestion
	height: ${({ deviceType }) => (deviceType === 'desktop' ? '24px' : '36px')};
	width: ${({ deviceType }) => (deviceType === 'desktop' ? '24px' : '36px')};
```

File: src/vimeo/players/shared/CloseButton.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+`;
+
+interface CloseButtonProps extends VimeoArticleVideoPlayerTrackingProps {
+	dismiss: () => void;
+}
+
+const CloseButton: React.FC<CloseButtonProps> = ({ dismiss, deviceType }) => {
+	const onClickClose = () => {
+		dismiss();
+	};
```
Comment (pbartkowiak-fandom) [inline src/vimeo/players/shared/CloseButton.tsx:34]: I think the intermediate `onClickClose` function isn't necessary and you could pass `dismiss` directly to the `CloseWrapper` component.

File: src/vimeo/players/shared/VimeoPlayerWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+			const script = document.createElement('script');
+			script.src = 'https://player.vimeo.com/api/player.js';
+			script.onload = () => {
+				console.debug('Vimeo script loaded.');
+				initVimeoPlayer();
+			};
+
+			const firstScriptTag = document.getElementsByTagName('script')[0];
+			firstScriptTag.parentNode.insertBefore(script, firstScriptTag);
+			document.getElementsByTagName('head')[0].appendChild(script);
```
[the remark this answers is not in the document]
Comment (nandy-andy) [reply src/vimeo/players/shared/VimeoPlayerWrapper.tsx:56]: Hm, I've happy-path tested it for mobile and desktop and it loads only once in my network tab 🤔

**Review by dzawada — APPROVED** [review]

**Review by pbartkowiak-fandom — APPROVED** [review]

File: src/loaders/utils/GetMediaWikiConfigDetails.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+export function getMediaWikiConfigDetails() {
+	const config = window?.mw?.config;
+	const wikiId = config?.get('wgCityId');
+	const isTier3Wiki = config?.get('wgArticleFeaturedVideo')?.tier3Mapping ?? false;
+
+	return {
+		wikiId,
+		isTier3Wiki,
+	};
```
Comment (pbartkowiak-fandom) [inline src/loaders/utils/GetMediaWikiConfigDetails.ts:13]: Please specify the return type. Currently, both `wikiId` and `isTier3Wiki` are seen as `any` type:

image

File: src/vimeo/players/VimeoDesktopArticleVideoPlayer.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+const VimeoDesktopArticleVideoTopPlaceholder = styled.div`
+	position: absolute;
+	width: 100%;
+	padding-top: 56.25%;
+	top: 0;
+	left: 0;
+	bottom: 0;
+	right: 0;
+	z-index: 2;
```
Comment (pbartkowiak-fandom) [inline src/vimeo/players/VimeoDesktopArticleVideoPlayer.tsx:18]: I wonder, why not use `WDSVariables` for that `z-index` value?

Comment (nandy-andy) [reply src/vimeo/players/VimeoDesktopArticleVideoPlayer.tsx:18]: That's a good question ;-) It was "heavily inspired" by Youtube Takeover, so I'll try out using the WDSVariables 👍

**Review by pbartkowiak-fandom — APPROVED** [review]

**Pull request merged**